### ISSUE TITLE
Please help me with the installation method of mac OS system？

### ISSUE THREAD

**Issue by YangHuaidong** [eclipse-ecal/ecal#1870]
### Context

rt

### Proposal

r t

**Comment by FlorianReimold** [comment]
macOS is very experimental. You need to compile eCAL yourself. You can use the CMake Options from the following workflow file:
https://github.com/eclipse-ecal/ecal/blob/master/.github/workflows/build-macos.yml

**Comment by YangHuaidong** [comment]
Does it mean that it doesn't support Mac system yet?

**Comment by YangHuaidong** [comment]
Can you help me pack a package for mac M2 system?

**Comment by YangHuaidong** [comment]
> macOS is very experimental. You need to compile eCAL yourself. You can use the CMake Options from the following workflow file: https://github.com/eclipse-ecal/ecal/blob/master/.github/workflows/build-macos.yml

Excuse me, I want to ask how to use this

**Comment by FlorianReimold** [comment]
Yes, as [written in the readme](https://github.com/eclipse-ecal/ecal/tree/master?tab=readme-ov-file#facts-about-ecal), macOS Support is experimental. We don't own macs, so there is no acceptable way for us to test on macOS.
eCAL does compile on macOS though, the GH Actions prove that. Not sure, what version of Apple Silicon currently is availabe, but it is some ARM based system nowadays.
The packaging of macOS binaries currently fails on the GH Actions. So we disabled it. The artifacts have never been tested anyways and probably never worked, so it's not a big loss. If you want to use eCAL on macOS, you have to compile it yourself. You can find all commands on how to do that in the [macOS GH Action file](https://github.com/eclipse-ecal/ecal/blob/master/.github/workflows/build-macos.yml).
If you cannot get eCAL to compile, you are probably better off using another middleware anyways, as you may have to actually put some development activities into eCAL to get it to work on macOS.

----

That being said, you can setup an Ubuntu VM on your mac and then [get eCAL precompiled via `apt-get`](https://eclipse-ecal.github.io/ecal/getting_started/setup.html#fa-ubuntu-automatically-install-ecal-from-a-ppa) (we have ARM builds available as well)
